### PR TITLE
fix: Implement retaining rewards for transfers

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
Closes: #7776

## What is the purpose of the change

This PR removes reward claiming logic that previously executing prior to position transfers. Since the transfers are done through manual reassignment of ownership over position IDs, handling forfeited incentives become tricky/hacky since the positions do not actually move (see issue for further discussion).

Spread reward claiming is also removed for consistency. This also makes the transfer function run more efficiently, as these claims were not necessary for the correct functioning of the logic.

## Testing and Verifying
Tests in `position_test.go` are updated to ensure that claiming of old and new rewards (both incentives and spread rewards) function properly (i.e. go to the new owner and not the old).

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented?
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

## Summary by CodeRabbit

- **Bug Fixes**
	- Updated event trigger counts in tests to accurately reflect no additional events being triggered during position transfers.
- **New Features**
	- Removed the ability to claim rewards during position transfers for enhanced system integrity and fairness.

### CODE REVIEW

File: x/concentrated-liquidity/position_test.go [hunk 1]
```diff
@@ -2440,31 +2440,34 @@ func (s *KeeperTestSuite) TestTransferPositions() {
...
-				expectedTransferOriginalOwnerFunds := preTransferOwnerFunds.Add(totalExpectedRewards...)
-				s.Require().Equal(expectedTransferOriginalOwnerFunds.String(), postTransferOriginalOwnerFunds.String())
+				s.Require().Equal(preTransferOwnerFunds.String(), postTransferOriginalOwnerFunds.String())
 
-				// Check that the new owner does not have any new funds
+				// Check that the new owner's balance did not change due to the transfer
 				postTransferNewOwnerFunds := s.App.BankKeeper.GetAllBalances(s.Ctx, newOwner)
 				s.Require().Equal(preTransferNewOwnerFunds, postTransferNewOwnerFunds)
 
-				// Test that claiming incentives/spread rewards with the new owner returns nothing
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/position_test.go:2452]: Seems like this diff is rendering weirdly. The changes here were:
1. Remove previous assertion that claiming rewards post transfer yields nothing
2. Claim rewards post transfer and ensure they go to new owner

**Review by czarcas7ic — APPROVED** [review]
ACK!

File: x/concentrated-liquidity/position.go [hunk 2]
```diff
@@ -704,14 +704,6 @@ func (k Keeper) transferPositions(ctx sdk.Context, positionIds []uint64, sender
 			return types.LockNotMatureError{PositionId: position.PositionId, LockId: lockId}
 		}
 
-		// Collect any outstanding incentives and rewards for the position.
-		if _, err := k.collectSpreadRewards(ctx, sender, positionId); err != nil {
```
Comment (p0mvn) [inline x/concentrated-liquidity/position.go:708]: Let's update method spec to remove:
```
// It then collects any outstanding incentives and rewards for the position,
```

**Review by p0mvn — APPROVED** [review]

File: x/concentrated-liquidity/position.go (hunk 2 above)
Comment (AlpinYukseloglu) [reply x/concentrated-liquidity/position.go:708]: Nice catch, updated in 8b0977fed54641554dcef683174d03d163fcccfe

**Review by PaddyMc — APPROVED** [review]
:bow:

**Pull request merged**